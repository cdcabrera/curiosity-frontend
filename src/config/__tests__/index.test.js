import { config } from '../index';

describe('Configuration', () => {
  it('should have specific config properties', () => {
    expect(Object.keys(config)).toMatchSnapshot('config');

    // expect(Object.keys(config.products)).toMatchSnapshot('products');
    // expect(config.rbac).toMatchSnapshot('rbac');
    // expect(config.routes.map(obj => ({ ...Object.entries(obj).map }))).toMatchSnapshot('routes');
  });

  it('should have consistent product configuration', () => {
    expect(Object.keys(config.products)).toMatchSnapshot('products');

    // loop through configs and confirm they have a min number of the same properties
    // it('should set basic product configuration', () => {
    //  expect(config).toMatchSnapshot('initial configuration');
    // });
    // it('should set basic product configuration', () => {
    //     expect(config).toMatchSnapshot('initial configuration');
    //   });
  });

  it('should have a consistent rbac configuration', () => {
    Object.values(config.rbac).forEach(({ permissions }) => {
      expect(Array.isArray(permissions)).toBe(true);

      permissions.forEach((value, index) => {
        expect(`${index}, ${Object.keys(permissions[index]).join(', ')}`).toBe(`${index}, resource, operation`);
      });
    });
  });

  it('should have a consistent route configuration', () => {
    expect(Array.isArray(config.routes)).toBe(true);

    const inconsistentRoutes = [];

    config.routes.forEach((value, index) => {
      const entryCheck = {
        redirect:
          typeof config.routes[index].redirect === 'string' || config.routes[index].redirect === null ? 'PASS' : 'FAIL',
        isSearchable: typeof config.routes[index].isSearchable === 'boolean' ? 'PASS' : 'FAIL',
        aliases: Array.isArray(config.routes[index].aliases) ? 'PASS' : 'FAIL',
        activateOnError: typeof config.routes[index].activateOnError === 'boolean' ? 'PASS' : 'FAIL',
        disabled: typeof config.routes[index].disabled === 'boolean' ? 'PASS' : 'FAIL',
        default: typeof config.routes[index].default === 'boolean' ? 'PASS' : 'FAIL',
        component:
          typeof config.routes[index].component === 'string' || config.routes[index].component === null
            ? 'PASS'
            : 'FAIL'
      };

      if (Object.values(entryCheck).indexOf('FAIL') > -1) {
        inconsistentRoutes.push({ [`Route ${index} inconsistent`]: entryCheck });
      }
    });

    expect(inconsistentRoutes).toMatchSnapshot('inconsistent routes');
  });
});
