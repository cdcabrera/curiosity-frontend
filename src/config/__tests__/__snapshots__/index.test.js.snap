// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration should have specific config properties: config 1`] = `
Object {
  "rbac": Object {
    "inventory": Object {
      "permissions": Array [
        Object {
          "operation": "*",
          "resource": "*",
        },
        Object {
          "operation": "*",
          "resource": "hosts",
        },
        Object {
          "operation": "read",
          "resource": "hosts",
        },
        Object {
          "operation": "write",
          "resource": "hosts",
        },
      ],
    },
    "subscriptions": Object {
      "permissions": Array [
        Object {
          "operation": "*",
          "resource": "*",
        },
        Object {
          "operation": "*",
          "resource": "reports",
        },
        Object {
          "operation": "read",
          "resource": "reports",
        },
        Object {
          "operation": "write",
          "resource": "reports",
        },
      ],
    },
  },
  "routes": Array [
    Object {
      "activateOnError": false,
      "aliases": Array [
        "insights",
      ],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "rhel",
      "isSearchable": true,
      "path": "/rhel",
      "pathParameter": Array [
        "RHEL",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "RHEL",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "rhel-arm",
      "isSearchable": false,
      "path": "/rhel-arm",
      "pathParameter": Array [
        "RHEL for ARM",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "RHEL",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "rhel-ibmpower",
      "isSearchable": false,
      "path": "/rhel-ibmpower",
      "pathParameter": Array [
        "RHEL for IBM Power",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "RHEL",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "rhel-ibmz",
      "isSearchable": false,
      "path": "/rhel-ibmz",
      "pathParameter": Array [
        "RHEL for IBM z",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "RHEL",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "rhel-x86",
      "isSearchable": false,
      "path": "/rhel-x86",
      "pathParameter": Array [
        "RHEL for x86",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "RHEL",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewOpenShiftContainer",
      "default": false,
      "disabled": false,
      "id": "openshift-container",
      "isSearchable": true,
      "path": "/openshift-container",
      "pathParameter": Array [
        "OpenShift Container Platform",
        "OpenShift-metrics",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "cores",
              "isOptional": true,
              "stroke": "#06c",
            },
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "sockets",
              "isOptional": true,
              "stroke": "#06c",
            },
            Object {
              "id": "thresholdSockets",
              "isOptional": true,
            },
            Object {
              "id": "thresholdCores",
              "isOptional": true,
            },
          ],
          "initialGraphSettings": Object {},
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"sockets_OpenShift Container Platform\\"})",
              "id": "sockets",
              "isOptional": true,
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cellWidth": 15,
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"cores_OpenShift Container Platform\\"})",
              "id": "cores",
              "isOptional": true,
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 25,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialOption": "cores",
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "productContextFilterUom": true,
          "productId": "OpenShift Container Platform",
          "productLabel": "OpenShift",
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
            "uom": "cores",
          },
          "viewId": "viewOpenShift",
        },
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "coreHours",
              "stroke": "#06c",
            },
          ],
          "initialGraphSettings": Object {
            "actionDisplay": [Function],
          },
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "coreHours",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 25,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": undefined,
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "productId": "OpenShift-metrics",
          "productLabel": "OpenShift Metric",
          "query": Object {
            "beginning": "2019-07-01T00:00:00.000Z",
            "ending": "2019-07-31T23:59:59.999Z",
          },
          "viewId": "viewOpenShiftMetric",
        },
      ],
      "productParameter": Array [
        "OpenShift Container Platform",
        "OpenShift-metrics",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewOpenShiftDedicated",
      "default": false,
      "disabled": false,
      "id": "openshift-dedicated",
      "isSearchable": true,
      "path": "/openshift-dedicated",
      "pathParameter": Array [
        "OpenShift-dedicated-metrics",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "coreHours",
              "stroke": "#06c",
            },
          ],
          "initialGraphSettings": Object {
            "actionDisplay": [Function],
          },
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "coreHours",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"lastSeen_OpenShift-dedicated-metrics\\"})",
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": undefined,
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "query": Object {
            "beginning": "2019-07-01T00:00:00.000Z",
            "ending": "2019-07-31T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "OpenShift-dedicated-metrics",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "satellite",
      "isSearchable": false,
      "path": "/satellite",
      "pathParameter": Array [
        "Satellite",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "Satellite",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "satellite-capsule",
      "isSearchable": false,
      "path": "/satellite-capsule",
      "pathParameter": Array [
        "Satellite Capsule",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "Satellite",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewDefault",
      "default": false,
      "disabled": false,
      "id": "satellite-server",
      "isSearchable": false,
      "path": "/satellite-server",
      "pathParameter": Array [
        "Satellite Server",
      ],
      "productConfig": Array [
        Object {
          "graphTallyQuery": Object {
            "granularity": "daily",
          },
          "initialGraphFilters": Array [
            Object {
              "color": "#06c",
              "fill": "#8bc1f7",
              "id": "physicalSockets",
              "stroke": "#06c",
            },
            Object {
              "color": "#009596",
              "fill": "#a2d9d9",
              "id": "hypervisorSockets",
              "stroke": "#009596",
            },
            Object {
              "color": "#5752d1",
              "fill": "#b2b0ea",
              "id": "cloudSockets",
              "stroke": "#5752d1",
            },
            Object {
              "id": "thresholdSockets",
            },
          ],
          "initialGuestsFilters": Array [
            Object {
              "cell": [Function],
              "header": "t(curiosity-inventory.header, {\\"context\\":\\"guestsDisplayName\\"})",
              "id": "displayName",
            },
            Object {
              "cellWidth": 40,
              "id": "inventoryId",
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
            },
          ],
          "initialInventoryFilters": Array [
            Object {
              "cell": [Function],
              "id": "displayName",
              "isSortable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 20,
              "id": "measurementType",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "sockets",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "lastSeen",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialInventorySettings": Object {},
          "initialSubscriptionsInventoryFilters": Array [
            Object {
              "id": "productName",
              "isSortable": true,
            },
            Object {
              "cellWidth": 15,
              "id": "serviceLevel",
              "isSortable": true,
              "isWrappable": true,
            },
            Object {
              "cell": [Function],
              "cellWidth": 15,
              "id": "upcomingEventDate",
              "isSortable": true,
              "isWrappable": true,
            },
          ],
          "initialToolbarFilters": Array [
            Object {
              "id": "sla",
            },
            Object {
              "id": "usage",
              "selected": true,
            },
          ],
          "inventoryHostsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "last_seen",
          },
          "inventorySubscriptionsQuery": Object {
            "dir": "desc",
            "limit": 100,
            "offset": 0,
            "sort": "upcoming_event_date",
          },
          "query": Object {
            "beginning": "2019-06-20T00:00:00.000Z",
            "ending": "2019-07-20T23:59:59.999Z",
          },
        },
      ],
      "productParameter": Array [
        "Satellite",
      ],
      "redirect": null,
    },
    Object {
      "activateOnError": true,
      "aliases": Array [],
      "component": "optinView/optinView",
      "default": false,
      "disabled": false,
      "id": "optin",
      "isSearchable": false,
      "path": "/optin",
      "pathParameter": null,
      "productConfig": null,
      "productParameter": null,
      "redirect": null,
    },
    Object {
      "activateOnError": false,
      "aliases": Array [],
      "component": "productView/productViewMissing",
      "default": true,
      "disabled": false,
      "id": "missing",
      "isSearchable": false,
      "path": "/",
      "pathParameter": null,
      "productConfig": null,
      "productParameter": null,
      "redirect": "/",
    },
  ],
}
`;
